pkg/git/repo.go

```diff
@@ -486,5 +486,9 @@ func (r *Repo) workingClone(ctx context.Context, ref string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return clone(ctx, working, r.dir, ref)
+	path, err := clone(ctx, working, r.dir, ref)
+	if err != nil {
+		os.RemoveAll(path)
```

Comment:
It's a quibble, but this ought to be

    os.RemoveAll(workingDir)

since that is the directory created here. At one point `clone` created a directory _under_ the working dir it was given, now it returns the same directory -- if it did something like that again, this would start leaking directories.

It's probably better to remove the possibility of it being a different directory and change `clone` to return only an error, so there's no mistaking which path is used. But for now: removing the `workingDir` guards against that regression.

Reply:
Ah, good catch! Thanks